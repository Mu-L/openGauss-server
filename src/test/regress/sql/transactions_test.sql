CREATE TABLE no_reset_test(a int);
insert into no_reset_test values(1);

-- Test transaction variable cannot be reset.
BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
select count(*) from no_reset_test;
reset transaction_isolation; --error
END;

BEGIN TRANSACTION READ ONLY;
select count(*) from no_reset_test;
reset transaction_read_only; --error
END;

BEGIN TRANSACTION DEFERRABLE;
select count(*) from no_reset_test;
reset transaction_deferrable; --error
END;

DROP TABLE no_reset_test;

SET SESSION CHARACTERISTICS AS TRANSACTION READ ONLY;
START TRANSACTION READ WRITE;
CREATE TABLE transaction_test_table(a int);
insert into transaction_test_table values(1);
update transaction_test_table set a=2 where a=1;
select * from transaction_test_table;
commit;
-- error
delete from transaction_test_table;
drop table transaction_test_table;

START TRANSACTION READ WRITE;
drop table transaction_test_table;
commit;
