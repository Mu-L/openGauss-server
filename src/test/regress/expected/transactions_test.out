CREATE TABLE no_reset_test(a int);
insert into no_reset_test values(1);
-- Test transaction variable cannot be reset.
BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
select count(*) from no_reset_test;
 count 
-------
     1
(1 row)

reset transaction_isolation; --error
ERROR:  cannot reset parameter "transaction_isolation"
END;
BEGIN TRANSACTION READ ONLY;
select count(*) from no_reset_test;
 count 
-------
     1
(1 row)

reset transaction_read_only; --error
ERROR:  cannot reset parameter "transaction_read_only"
END;
BEGIN TRANSACTION DEFERRABLE;
select count(*) from no_reset_test;
 count 
-------
     1
(1 row)

reset transaction_deferrable; --error
ERROR:  cannot reset parameter "transaction_deferrable"
END;
DROP TABLE no_reset_test;
SET SESSION CHARACTERISTICS AS TRANSACTION READ ONLY;
START TRANSACTION READ WRITE;
CREATE TABLE transaction_test_table(a int);
insert into transaction_test_table values(1);
update transaction_test_table set a=2 where a=1;
select * from transaction_test_table;
 a 
---
 2
(1 row)

commit;
-- error
delete from transaction_test_table;
ERROR:  cannot execute DELETE in a read-only transaction
drop table transaction_test_table;
ERROR:  cannot execute DROP TABLE in a read-only transaction
START TRANSACTION READ WRITE;
drop table transaction_test_table;
commit;
